templates/terraform/examples/app_engine_standard_app_version.tf.erb

```diff
@@ -0,0 +1,38 @@
+resource "google_project" "my_project" {
+  name       = "tf-test-project"
+  project_id = "<%= ctx[:vars]['project_id'] %>"
+  org_id     = "<%= ctx[:test_env_vars]['org_id'] %>"
+}
+
+resource "google_app_engine_application" "app" {
+  project     = "${google_project.my_project.project_id}"
+  location_id = "us-central"
+}
+
+resource "google_storage_bucket" "bucket" {
+	name = "<%= ctx[:vars]['bucket_name'] %>"
+}
+
+resource "google_storage_bucket_object" "object" {
+	name   = "hello-world.zip"
+	bucket = "${google_storage_bucket.bucket.name}"
+	source = "./test-fixtures/appengine/hello-world.zip"
+}
+
+resource "google_app_engine_standard_app_version" "<%= ctx[:primary_resource_id] %>" {
+  version_id = "v1"
+  service = "<%= ctx[:vars]['service_name'] %>"
```

Comment:
This value has to be `"default"`, right?

Settings the value in `vars` is used for values with variable values, particularly resource names. Since `service` must be `"default"`, it can be hardcoded.

Reply:
Running the test, I receive this error. Sounds like this is configurable, but it may be problematic to use a different service in CI?

```
------- Stdout: -------
=== RUN   TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
=== PAUSE TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
=== CONT  TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
--- FAIL: TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample (29.83s)
    testing.go:568: Step 0 error: errors during apply:
        
        Error: Error creating StandardAppVersion: googleapi: Error 400: The first service (module) you upload to a new application must be the 'default' service (module). Please upload a version of the 'default' service (module) before uploading a version for the 'tf-test-servicew4qvwexmrl' service (module). See the documentation for more information. Python: (https://developers.google.com/appengine/docs/python/modules/#Python_Uploading%%20modules) Java: (https://developers.google.com/appengine/docs/java/modules/#Java_Uploading%%20modules)
        
          on /opt/teamcity-agent/temp/buildTmp/tf-test140929662/main.tf line 23:
          (source code not available)
        
        
FAIL
```

Reply:
service is configurable for the resource, I have tested locally that if a non-existence service is provided then it is created and a new version is uploaded. But it appears that if a new project is created then we must upload a version to default service. 
I have reverted the example to use the "default" service. And will test today .

Reply:
I am trying hard to get the tests passed, not sure why they keep failing, now I am not creating a project instead I am just adding the version to service default.
```
2019/08/15 10:28:15 [DEBUG] Starting graph walk: walkPlanDestroy
--- FAIL: TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample (47.44s)
    testing.go:568: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: google_app_engine_standard_app_version.version_id
          deployment.#:                  "0" => "1"
          deployment.0.files.#:          "" => "0"
          deployment.0.zip.#:            "" => "1"
          deployment.0.zip.0.source_url: "" => "https://storage.googleapis.com/appengine-static-contentk6a47sozsz/hello-world.zip"
          entrypoint.#:                  "0" => "1"
          entrypoint.0.shell:            "" => "node ./app.js"
          env_variables.port:            "" => "8080"
          handlers.#:                    "0" => "0"
          id:                            "apps/my-project-1540699643621/services/default/versions/v2" => "apps/my-project-1540699643621/services/default/versions/v2"
          libraries.#:                   "0" => "0"
          name:                          "apps/my-project-1540699643621/services/default/versions/v2" => "apps/my-project-1540699643621/services/default/versions/v2"
          project:                       "my-project-1540699643621" => "my-project-1540699643621"
          runtime:                       "nodejs10" => "nodejs10"
          runtime_api_version:           "" => ""
          service:                       "default" => "default"
          threadsafe:                    "true" => ""
          version_id:                    "v2" => "v2"
        
        
        
        STATE:
        
        google_app_engine_standard_app_version.version_id:
          ID = apps/my-project-1540699643621/services/default/versions/v2
          provider = provider.google
          name = apps/my-project-1540699643621/services/default/versions/v2
          project = my-project-1540699643621
          runtime = nodejs10
          runtime_api_version = 
          service = default
          threadsafe = true
          version_id = v2
        
          Dependencies:
            google_storage_bucket.bucket
        google_storage_bucket.bucket:
          ID = appengine-static-contentk6a47sozsz
          provider = provider.google
          bucket_policy_only = false
          force_destroy = false
          location = US
          name = appengine-static-contentk6a47sozsz
          project = my-project-1540699643621
          requester_pays = false
          self_link = https://www.googleapis.com/storage/v1/b/appengine-static-contentk6a47sozsz
          storage_class = STANDARD
          url = gs://appengine-static-contentk6a47sozsz
        google_storage_bucket_object.object:
          ID = appengine-static-contentk6a47sozsz-hello-world.zip
          provider = provider.google
          bucket = appengine-static-contentk6a47sozsz
          cache_control = 
          content_disposition = 
          content_encoding = 
          content_language = 
          content_type = application/zip
          crc32c = NOB6xA==
          detect_md5hash = M4k7M6WJTeuAuQw0qaPiDw==
          md5hash = M4k7M6WJTeuAuQw0qaPiDw==
          name = hello-world.zip
          output_name = hello-world.zip
          self_link = https://www.googleapis.com/storage/v1/b/appengine-static-contentk6a47sozsz/o/hello-world.zip
          source = ./test-fixtures/appengine/hello-world.zip
          storage_class = STANDARD
        
          Dependencies:
            google_storage_bucket.bucket
FAIL
FAIL	github.com/terraform-providers/terraform-provider-google/google	47.471s
GNUmakefile:16: recipe for target 'testacc' failed
```

Reply:
@rileykarson Any feedback on this. The log shows that all the resources are properly deleted. The service version GET call return 404 which means that version is deleted and I also verified it from console, both bucket object (hello-world.zip) and bucket also get deleted.
So I am not sure why the plan is not empty.

Reply:
Terraform will fail a test step if there's a non-empty plan after applying a config. We're getting one because some fields are not returned in response bodies. Upon seeing no value for a field in the response, Terraform will unset them (as you can see in plan)

We're performing a Create with the following request:

```
{
    "deployment": {
        "zip": {
            "sourceUrl": "https://storage.googleapis.com/appengine-static-contentswmz4pxf12/hello-world.zip"
        }
    },
    "entrypoint": {
        "shell": "node ./app.js"
    },
    "envVariables": {
        "port": "8080"
    },
    "id": "v2",
    "runtime": "nodejs10"
}
```

But getting back this:

```
{
    "createdBy": "{{serviceAccount}}",
    "createTime": "2019-08-16T16:42:36Z",
    "diskUsageBytes": "3373",
    "env": "standard",
    "id": "v2",
    "instanceClass": "F1",
    "name": "apps/{{project}}/services/default/versions/v2",
    "network": {
    },
    "runtime": "nodejs10",
    "runtimeChannel": "default",
    "servingStatus": "SERVING",
    "threadsafe": true,
    "versionUrl": "https://v2-dot-{{project}}.appspot.com"
}
```

As you can see in the Terraform diff and req/response pair, `deployment`, `entrypoint`, and `env_variables` aren't present.

You can set [`ignore_read`](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/overrides/terraform/property_override.rb#L46-L48) on those fields in `terraform.yaml` so that Terraform doesn't attempt to set them on reads.

Reply:
@rileykarson Once again thank you very much, I have added the ignore_read flag for deployment and entrypoint fields and for the first time I got integration tests passing horrah. :)
root@dadce36c0bce:/mm# ./integration_tests.sh
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample -timeout 240m -ldflags="-X=
github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
=== PAUSE TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
=== CONT  TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample
--- PASS: TestAccAppEngineStandardAppVersion_appEngineStandardAppVersionExample (48.42s)
PASS
ok      github.com/terraform-providers/terraform-provider-google/google 48.435s